binding/go/rhino.go

```diff
@@ -206,39 +217,48 @@ func (rhino *Rhino) Init() error {
 
 	if _, err := os.Stat(rhino.ModelPath); os.IsNotExist(err) {
 		return &RhinoError{
-			INVALID_ARGUMENT,
-			fmt.Sprintf("Specified model file could not be found at %s", rhino.ModelPath)}
+			StatusCode: INVALID_ARGUMENT,
+			Message: 	fmt.Sprintf("Specified model file could not be found at %s", rhino.ModelPath)}
 	}
 
 	if rhino.ContextPath == "" {
 		return &RhinoError{
-			INVALID_ARGUMENT,
-			"No valid context was provided"}
+			StatusCode: INVALID_ARGUMENT,
+			Message: 	"No valid context was provided"}
 	}
 
 	if _, err := os.Stat(rhino.ContextPath); os.IsNotExist(err) {
 		return &RhinoError{
-			INVALID_ARGUMENT,
-			fmt.Sprintf("Context file could not be found at %s", rhino.ContextPath)}
+			StatusCode: INVALID_ARGUMENT,
+			Message: 	fmt.Sprintf("Context file could not be found at %s", rhino.ContextPath)}
 	}
 
 	if rhino.Sensitivity < 0 || rhino.Sensitivity > 1 {
 		return &RhinoError{
-			INVALID_ARGUMENT,
-			fmt.Sprintf("Sensitivity value of %f is invalid. Must be between [0, 1]", rhino.Sensitivity)}
+			StatusCode: INVALID_ARGUMENT,
+			Message: 	fmt.Sprintf("Sensitivity value of %f is invalid. Must be between [0, 1]", rhino.Sensitivity)}
 	}
 
 	if rhino.EndpointDurationSec < 0.5 || rhino.EndpointDurationSec > 5.0 {
 		return &RhinoError{
-			INVALID_ARGUMENT,
-			fmt.Sprintf("EndpointDurationSec value of %f is invalid. Must be between [0.5, 5.0]", rhino.EndpointDurationSec)}
+			StatusCode: INVALID_ARGUMENT,
+			Message:	fmt.Sprintf("EndpointDurationSec value of %f is invalid. Must be between [0.5, 5.0]", rhino.EndpointDurationSec)}
 	}
 
 	status := nativeRhino.nativeInit(rhino)
 	if PvStatus(status) != SUCCESS {
+		errorStatus, messageStack := nativeRhino.nativeGetErrorStack()
+		if errorStatus != SUCCESS {
+			return &RhinoError{
+				StatusCode:   errorStatus,
+				Message:      "Unable to get Rhino error state",
+			}
+		}
+
 		return &RhinoError{
-			PvStatus(status),
-			"Rhino init failed"}
+			StatusCode:		PvStatus(status),
+			Message:		"Rhino init failed",
+			MessageStack: 	messageStack}
```

Comment:
is all this weird spacing Go's formatter? If so, it's fine, I'm just surprised how random it looks at time